### DISCUSSION TITLE
Custom Libki Helper app for Autopilot Kiosk

### DISCUSSION THREAD

**Discussion by MartinJLittle** [Libki/libki-server#313] [Show and tell]
I have recently completed developing an Autopilot delivered, intune managed Kiosk using Libki.

Due to the way that Multi-App kiosk settings work the supplied secondary "helper" apps (on-startup, on_login & on_logout) do not function correctly. (Or I could not get them working at least.)

Multi-App Kiosk uses Applocker in the background and launches into a full-screen start menu rather than a desktop, the start menu obscures the Libki client regardless of the window OnTop setting and the helper apps call a lot of EXEs in System32 which requires whitelisting those. I was reticent about allowing access to CMD.exe in particular as this would potentially allow users to run batch files from a USB drive.

With this in mind I went about creating my own helper app in AutoIT as I have a fair amount of experience in that language and it has a good amount of window management and inbuilt functions to avoid using external EXEs.

I have attached the source code if anyone else wants to take a look.

My custom install of Libki:
Launches the libki installer silently
Renames the 3 helper apps post install.
Copies my custom helper to the "C:\Program Files (x86)\Libki\Windows" folder.
Removes the registry key for Libki from "HKLM:\Software\Microsoft\Windows\CurrentVersion\Run".
Creates a scheduled task to run Libki on login instead (I prefer Scheduled Tasks).
Edits the INI with custom settings for my environment including extracting the site from the computername since our naming convention includes the sitename. (If the name doesn't include the sitename, then it allocates the device to our test site instead.)

If this benefits anyone else, I'll be very happy.

Martin

[LibkiHelper.zip](https://github.com/Libki/libki-server/files/11905394/LibkiHelper.zip)

**Comment by kylemhall** [comment]
Fantastic! I can't wait to have a look at your work!
